Addressing Issue: Temporary failure in name resolution

edavalosanaya opened the pull request:
This PR is made to address #82. It is composed of three commits: 

- First, added ``dotenv`` to the ``__init__.py`` of the ``g3pylib`` package to ensure that the ``G3_HOSTNAME`` environmental variable is loaded when the package is loaded. Not sure if this is the desired behavior, as the examples all perform this function within their mains. Having this as default would benefit the tests, as the loading of the environmental variables was not performed, causing certain tests to fail. 
- Had to ignore certain tests that would hang and freeze. I couldn't run the complete test suite with ``pytest``. I added ``@pytest.mark.skip`` to these offending tests. Now the test suite should be able to run, yet it still has some failing tests. Also added a timeout for tests.
- The ``streamgazerstp.py`` example felt lacking in that it would not show the gaze on the video. Just made OpenCV draw the gaze on the video.

NOTES: 
Used ``pre-commit``.

Best,
Eduardo Davalos

edavalosanaya commented:
Just realized that you are using ``pytest-dotenv`` plugin. Does that mean that ``.env`` needed to be store within the ``test`` folder?

jonashogstrom commented on examples/streamgazertsp.py:
```diff
@@ -26,18 +26,25 @@ async def stream_rtsp():
                         logging.info(f"Gaze2d: {gaze2d[0]:9.4f},{gaze2d[1]:9.4f}")
+                        h, w = img.shape[:2]
+                        fix = (int(gaze2d[0] * w), int(gaze2d[1] * h))
+                        img = cv2.circle(img.copy(), fix, 10, (0, 0, 255), 3)
```
Nice addition with the gaze overlay. Is it really necessary to copy the image? isn't it better to just draw on the original image (without setting the img-variable again)? 

https://www.geeksforgeeks.org/python-opencv-cv2-circle-method/

jonashogstrom replied:
I tried locally to remove the copy and the assignment and it still worked like a charm.

edavalosanaya replied:
I think you are right, will make those changes.

jonashogstrom commented on pyproject.toml:
```diff
@@ -65,3 +65,6 @@ skip_gitignore = true
 testpaths = ["tests"]
+
+# Timeout
+faulthandler_timeout=30
```
Another nice addition. A failed test is a better test than a non-terminating test.

jonashogstrom commented on tests/api_components/test_recordings.py:
```diff
@@ -9,6 +9,7 @@ async def test_get_name(g3: Glasses3):
     assert await g3.recordings.get_name() == "recordings"
 
 
+@pytest.mark.skip(reason="Test causes pytest to freeze")
```
Does this test freeze even after the timeout was added?

edavalosanaya replied:
Yes, it would cause the test suite to hang and not continue.

jonashogstrom commented:
Thanks for your contribution. The gaze overlay and the timeout are nice changes and if you submit them as separate PRs I'd be happy to approve them. I need to refresh my memory on the details on the .env file and tests. stay tuned.

jonashogstrom commented:
> Just realized that you are using `pytest-dotenv` plugin. Does that mean that `.env` needed to be store within the `test` folder?

the .env-file should be in the root folder (mentioned in the readme.md). When you have this, there is no need to call env.load_dotenv() in the main library. Since a real client application is unlikely to expect the hostname to be specified like this it is better to call `dotenv.load_dotenv()` in the initialization of the examples.

edavalosanaya commented:
Sorry for the delay, winter break + project deadline. Thank you for your helpful comments. I have removed the ``dotenv`` usage within the ``__init__``, cleaned up the example code, and remove the modification to the test case. For some reason, I cannot run the test, likely due to an error in my network setup.

jonashogstrom reviewed (approved):
Looks good now. Thanks @edavalosanaya !

jonashogstrom commented:
There are some code formatting errors. I'll resolve them and get this fix merged.
